e2e/nomostest/port_forward.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nomostest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	"kpt.dev/configsync/e2e"
+	"kpt.dev/configsync/pkg/kinds"
+)
+
+type portForward struct {
```

Comment:
Can we move this to a sub-package and rename it PortForwarder or DeploymentProxy? "port forward" is more of a verb than a noun.

Reply:
If it's just GetDeploymentPod, you can move that to a subdir too. Make it return an error, then wrap it with one that fails in NT.

Reply:
Oh, it uses the nt.GetObject things too huh... maybe those also should be in a testutil package... but then they also use the debug log... sigh

Reply:
Yeah, this would require a fairly extensive refactor. We also would need to move the associated `WatchObject` functions to a subpackage.

Reply:
Is this feasible now?

Reply:
The change has been rebased and moved to a subpackage